static/app/utils/useTeams.tsx

```diff
@@ -273,16 +292,21 @@ function useTeams({limit, slugs, ids, provideUserTeams}: Options = {}) {
 
       const fetchedTeams = uniqBy([...store.teams, ...results], ({slug}) => slug);
 
-      // Only update the store if we have more items
-      if (fetchedTeams.length > store.teams.length) {
-        TeamActions.loadTeams(fetchedTeams);
+      if (search) {
+        // Only update the store if we have more items
+        if (fetchedTeams.length > store.teams.length) {
+          TeamActions.loadTeams(fetchedTeams);
+        }
+      } else {
+        // If we fetched a page of teams without a search query, add cursor data to the store
+        TeamActions.loadTeams(fetchedTeams, hasMore, nextCursor);
       }
 
       setState({
         ...state,
-        hasMore,
+        hasMore: hasMore && store.hasMore,
```

Comment:
Right, this indicates that if the store already has everything we've loaded everything.